src/components/map-viewer.jsx

```diff
@@ -237,6 +254,103 @@ const MapViewer = ({
                                         }}
                                     ></NetworkMapTab>
                                 </Box>
+
+                                <Box
+                                    style={{
+                                        width:
+                                            studyDisplayMode ===
+                                            STUDY_DISPLAY_MODE.DRAW
+                                                ? '20%'
+                                                : '0%',
+                                        height: '100%',
+                                    }}
+                                >
+                                    {studyDisplayMode ===
+                                        STUDY_DISPLAY_MODE.DRAW && (
+                                        <FilterCreationPanel
+                                            onSaveSelection={async (
```

Comment:
onSaveSelection has become very large , may be extract it into a 2 function, one for filter creation and another for Contingencys ??